src/models/projectsFactory.ts

```diff
@@ -59,25 +60,25 @@ export default class ProjectsFactory extends AbstractModelFactory<ProjectDBSchem
    * @param projectData - project data for creation
    */
   public async create(projectData: ProjectDBScheme): Promise<ProjectModel> {
-    const projectId = (await this.collection.insertOne(projectData)).insertedId;
+    const integrationId = uuid.v4();
+    const encodedIntegrationToken = ProjectModel.generateIntegrationToken(integrationId);
```

Comment:
why the `integrationId` is passed from the outside, but secret generated inside? Does it related to the project model?